File: strongly-typed-typescript/immutability/safe.ts
class Circle {
  readonly radius: number;
  constructor(radius: number) { this.radius = radius; }
  public get area() { return Math.PI * Math.pow(this.radius, 2); }
}

let unitCircle = new Circle(1.0);
unitCircle.radius = 2.0;
Math.PI = 0.0;

File: strongly-typed-go/immutability/circle/safe.go
package circle
import "math"

type Circle struct {
	r float64
}
func (c *Circle) Radius() float64 { return c.r }
func (c *Circle) Area() float64 { return math.Pi * c.r * c.r }
func NewCircle(radius float64) *Circle {
	return &Circle{r: radius}
}